### PR TITLE
[C-977] Add Screen change analytics

### PR DESCRIPTION
**Pull request by sliptype**
### Description

* Add analytics on screen change on mobile
* Primary routes are recorded via the NavigationContainer `onStateChange`
* Secondary routes are recorded via effects in the corresponding components
   This is necessary because in some cases the data needed to build the url is not available in the navigation state. Track screen for example records a url like this: `/KSHMRmusic/anywheres-home` but we only have track id available in the navigation state

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/screens/profile-screen/ProfileScreen.tsx [hunk 1]
```diff
@@ -104,6 +106,15 @@ export const ProfileScreen = () => {
     }
   }, [status])
 
+  // Record screen view
+  useEffect(() => {
```
Comment (dylanjeffers) [inline packages/mobile/src/screens/profile-screen/ProfileScreen.tsx:110]: just wondering why some are effect and some are focusEffect?

**Review by dylanjeffers — APPROVED** [review]
sweet great changes, lmk if you want to discuss putting it in screen component

File: packages/mobile/src/screens/profile-screen/ProfileScreen.tsx (hunk 1 above)
Comment (sliptype) [reply packages/mobile/src/screens/profile-screen/ProfileScreen.tsx:110]: I was using useEffect when the url is based on data inside the component and useFocusEffect when the url is a static value. Pushing this into the screen component means I can use useEffect across the board with a dependency on the url

File: packages/mobile/src/components/navigation-container/NavigationContainer.tsx [hunk 2]
```diff
@@ -157,7 +164,27 @@ const NavigationContainer = (props: NavigationContainerProps) => {
...
   return (
-    <RNNavigationContainer linking={linking} theme={navigationThemes[theme]}>
+    <RNNavigationContainer
+      linking={linking}
+      theme={navigationThemes[theme]}
+      ref={navigationRef}
+      onReady={() => {
+        routeNameRef.current = getPrimaryRoute(navigationRef.getRootState())
+      }}
+      onStateChange={async () => {
```
[the remark this answers is not in the document]
Comment (sliptype) [reply packages/mobile/src/components/navigation-container/NavigationContainer.tsx]: I double checked and these aren't causing any extra renders. Also this is how the docs recommend to do it: https://reactnavigation.org/docs/screen-tracking/

**Review by dylanjeffers — APPROVED** [review]
awesome changes

**Pull request merged**